### ISSUE TITLE
Deprecate UnconstrainedProblems / add getproblem API

### ISSUE THREAD

**Issue by anriseth** [JuliaNLSolvers/OptimTestProblems.jl#10]
Ref https://github.com/JuliaNLSolvers/OptimTestProblems.jl/pull/9#issuecomment-366154267

**Comment by pkofod** [comment]
Yes, that's exactly what I mean. If you tag this package now, Optim will break, so you have to put an upper bound on Optim and that's annoying. Or am I misunderstanding?

**Comment by anriseth** [comment]
It won't break, as I added backwards-compatibility in the PR. See, for example, https://github.com/JuliaNLSolvers/OptimTestProblems.jl/blob/master/src/OptimTestProblems.jl#L8. The annoying thing is that I don't know how to tell the user to stop using the legacy stuff.

**Comment by anriseth** [comment]
Just to be clear, the two things that I want to deprecate are
https://github.com/JuliaNLSolvers/OptimTestProblems.jl/blob/master/src/OptimTestProblems.jl#L8
https://github.com/JuliaNLSolvers/OptimTestProblems.jl/blob/master/src/optim_tests/multivariate/unconstrained.jl#L7

**Comment by pkofod** [comment]
Ah, perfect then! Anyway, we're the only users here. The lesson is to use something like `getproblem()` instead of dictionaries for stuff like this. If something **could** be deprecated at some point, it's always better to use a function.
